### ISSUE TITLE
Compiling TypeScript gets slower with every change

### ISSUE THREAD

**Issue by davelosert** [virtualidentityag/gulp-build-framework#22]
When using `yarn start` to start the dev-server, TypeScript-Compilation with Webpack gets exponentially slower with every change that triggers recompilation of typescript. 

So the first few changes are done within a few milliseconds, later on it suddenly takes several seconds.

I suspect there is some sort of memory-leak or something that consumes to many ressources.
I will investigate this further but thought to open an Issue so everybody is aware.

**Comment by janrembold** [comment]
i can confirm that :( 
@davelosert do you think it would be better to run webpack with its own watch task in parallelshell or something like that?

**Comment by davelosert** [comment]
@janrembold : Do you use the webpack-dev-server? That could be a good option...
if not I'd recommend for a quick fix to change to use webpack's own watch service to try to see if it fixes that.

**Comment by janrembold** [comment]
No, we currently use webpack-stream. I found out that both webpack tasks (React and TS) used the same webpack instance. That might have caused the memory issues. I also activated caching in webpack config. Maybe this also helps to tease out some performance.

I tagged a new version "4.2.0-rc2" with this update. It also contains a fix for those annoying multiple livereloads on SASS changes.

So please upgrade your dependency to "4.2.0-rc2" and try the new build framework. Hope that fixes this problem.

**Comment by davelosert** [comment]
@janrembold Sounds great, thanks a lot. I will try it out next week.

**Comment by janrembold** [comment]
@davelosert I just updated webpack, webpack-stream, typescript and ts-loader packages to newest version. We also have explicit webpack instances for React and TS in Version 4.2.0 now. https://github.com/virtualidentityag/gulp-build-framework/tree/4.2.0 Please test this version in heavy TS projects

**Comment by davelosert** [comment]
@janrembold As we changed to another build process for our application, I won't be able to check it, soon. But just fyi: I am having tghe same problem (memory overflow) with our new build system as well... seems to be something with webpack and typescript maybe? I don't know...

**Comment by timomayer** [comment]
Since we remove the react part from the frontend framework (v4.5 aka 5) and switch to a micro frontends apporach where every micro frontend has its own build process, can we close that?

**Comment by davelosert** [comment]
Would be okay for me since we don't currently have the need for a fix

**Comment by janrembold** [comment]
I still can't reproduce this behavior reliably. Maybe this got already fixed with the polling fix for the Windows watch task. I would like to keep it open until we have finally tested the new 5.0 version. the webpack-stream plugin that is mentioned here is still in use for TypeScript bundling

**Comment by timomayer** [comment]
@janrembold didnt we discuss to use gulp-typescript instead?
